File: src/main/java/com/github/novicezk/midjourney/dto/SubmitBlendDTO.java
package com.github.novicezk.midjourney.dto;

import com.github.novicezk.midjourney.enums.BlendDimensions;
import io.swagger.annotations.ApiModel;
import io.swagger.annotations.ApiModelProperty;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.List;

@Data
@ApiModel("Blend Submission Parameters")
@EqualsAndHashCode(callSuper = true)
public class SubmitBlendDTO extends BaseSubmitDTO {

	@ApiModelProperty(value = "Base64 array of images", required = true, example = "[\"data:image/png;base64,xxx1\", \"data:image/png;base64,xxx2\"]")
	private List<String> base64Array;

	@ApiModelProperty(value = "Aspect ratio: PORTRAIT(2:3); SQUARE(1:1); LANDSCAPE(3:2)", example = "SQUARE")
	private BlendDimensions dimensions = BlendDimensions.SQUARE;
}

File: src/dto/SubmitBlendDTO.ts
import { BaseSubmitDTO } from './BaseSubmitDTO';
import { BlendDimensions } from '../enums/BlendDimensions';

/**
 * Blend submission parameters
 */
export class SubmitBlendDTO extends BaseSubmitDTO {
  /** Base64 array of images */
  base64Array?: string[];

  /** Aspect ratio: PORTRAIT(2:3); SQUARE(1:1); LANDSCAPE(3:2) */
  dimensions: BlendDimensions = BlendDimensions.SQUARE;
}

